### PR TITLE
Added count exceeded warning

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary
Added user count exceeded warning

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-56357

#### Release Note
```release-note
None
```

### CODE REVIEW

File: server/channels/app/user.go
Comment (M-ZubairAhmed) [inline server/channels/app/user.go:330]: not a blocker but can this be improved instead of if inside of else?

**Review by M-ZubairAhmed — APPROVED** [review]
Is this the only possible flow of the user creation

File: server/channels/app/user.go
Comment (harshilsharma63) [reply server/channels/app/user.go:330]: Yes, everything comes to this function - signup, invite, SSO, everything.

File: server/channels/app/user.go
Comment (marianunez) [inline server/channels/app/user.go:328]: Here is a suggestion but let's wait for final confirmation from John when he gets back before we merge this 

```suggestion
			mlog.Warn("User created has exceeded the permissible total active users.", mlog.Int("user_limit", userLimits.MaxUsersLimit))
```

File: server/channels/app/user.go
Comment (marianunez) [reply server/channels/app/user.go]: How about the path to activate a user? Deactivate user -> Active user

**Review by marianunez — CHANGES_REQUESTED** [review]
Thanks @harshilsharma63!

File: server/channels/app/user.go
Comment (harshilsharma63) [reply server/channels/app/user.go]: Done.

**Review by marianunez — COMMENTED** [review]
Thanks @harshilsharma63! Code LGTM - will approve once we agree on final warning text

**Review by larkox — APPROVED** [review]

File: server/channels/app/user.go
Comment (harshilsharma63) [reply server/channels/app/user.go]: @marianunez forgot to comment, but I added check for this.

**Comment by harshilsharma63** [comment]
@marianunez I've updated the text.

File: server/channels/app/user.go
Comment (marianunez) [reply server/channels/app/user.go:327]: Perfect, thanks!

**Review by marianunez — APPROVED** [review]

**Pull request merged**
